ControlRoomApplication/ControlRoomApplication/Controllers/Communications/RemoteListener.cs

```diff
@@ -128,188 +217,619 @@ public bool RequestToKillTCPMonitoringRoutine()
+            return timeToMove;
+
+        }
+
+        public int RelativeMovementETA(Orientation movingBy)
+        {
+            // distance is degrees to steps for az/el
+            // time is in milliseconds
+            int EL_Speed = ConversionHelper.DPSToSPS(ConversionHelper.RPMToDPS(0.6), MotorConstants.GEARING_RATIO_ELEVATION);
+            int AZ_Speed = ConversionHelper.DPSToSPS(ConversionHelper.RPMToDPS(0.6), MotorConstants.GEARING_RATIO_AZIMUTH);
+
+            int positionTranslationAZ = ConversionHelper.DegreesToSteps(movingBy.Azimuth, MotorConstants.GEARING_RATIO_AZIMUTH);
+            int positionTranslationEL = ConversionHelper.DegreesToSteps(movingBy.Elevation, MotorConstants.GEARING_RATIO_ELEVATION);
+
+            int timeToMoveEl = MCUManager.EstimateMovementTime(EL_Speed, positionTranslationEL);
+            int timeToMoveAz = MCUManager.EstimateMovementTime(AZ_Speed, positionTranslationAZ);
+
+            // return the greater of the two times (we have to wait for longest mvmt)
+            int timeToMove = timeToMoveEl > timeToMoveAz ? timeToMoveEl : timeToMoveAz;
+            return timeToMove;
+        }
+
+        public int ScriptETA(string scriptType)
+        {
+            switch (scriptType.Trim())
+            {
+                case "DUMP":
+                    return 0;  
+
+                case "FULL_EV":
+                    return AbsoluteMovementETA(new Orientation(0, 90)) + RelativeMovementETA(new Orientation(0, -90));
+
+                case "THERMAL_CALIBRATE":
+                    return 0;
+
+                case "STOW":
+                    return AbsoluteMovementETA(new Orientation(0, 90));
+
+                case "FULL_CLOCK":
+                    return AbsoluteMovementETA(new Orientation(360, 0));
+
+                case "FULL_COUNTER":
+                    return AbsoluteMovementETA(new Orientation(-360,0));
+
+                case "HOME":
+                    return AbsoluteMovementETA(new Orientation(0, 0));
+
+                case "HARDWARE_MVMT_SCRIPT":
+                    return 0;
+
+                // If no command is found, invalid script. Return -1
+                default:
+                    return -1;
+            }
+            return -1;
+        }
+        /// <summary>
+        /// Used for when the mobile app sends a command for REQUEST | MVMT_DATA. This will specify the current location of the telescope, and whether or not
+        /// it is currently moving.
+        /// </summary>
+        /// <returns></returns>
+        public string GetMovementData()
+        {
+            Orientation currentPos = rtController.GetCurrentOrientation();
+            string currentlyMoving = rtController.RadioTelescope.PLCDriver.MotorsCurrentlyMoving().ToString().ToUpper();
+            return "MOVING: " + currentlyMoving + " | " + "AZ: " + currentPos.Azimuth + " | " + "EL: " + currentPos.Elevation;  
         }
     }
+
+    
+
+
+   
+   
```

Comment:
There is some whitespace down here that can probably be reduced.

- [ ] Consider reducing whitespace